include/inexor/vulkan-renderer/renderer.hpp

```diff
@@ -71,44 +70,30 @@ class VulkanRenderer {
 
     Camera m_game_camera;
 
-    // RAII wrapper for glfw contexts.
     std::unique_ptr<wrapper::GLFWContext> m_glfw_context;
-
-    // RAII wrapper for glfw windows.
     std::unique_ptr<wrapper::Window> m_window;
-
-    // RAII wrapper for VkInstance.
     std::unique_ptr<wrapper::Instance> m_vkinstance;
-
-    // RAII wrapper for VkDevice, VkPhysicalDevice and VkQueues.
     std::unique_ptr<wrapper::Device> m_vkdevice;
-
-    // RAII wrapper for glfw compatible Vulkan surfaces.
     std::unique_ptr<wrapper::WindowSurface> m_surface;
-
-    // RAII wrapper for Swapchain.
     std::unique_ptr<wrapper::Swapchain> m_swapchain;
-
-    // RAII wrapper for command pools.
     std::unique_ptr<wrapper::CommandPool> m_command_pool;
-
+    std::unique_ptr<imgui::ImGUIOverlay> m_imgui_overlay;
     std::unique_ptr<wrapper::Semaphore> m_image_available_semaphore;
     std::unique_ptr<wrapper::Semaphore> m_rendering_finished_semaphore;
+    std::unique_ptr<FrameGraph> m_frame_graph;
 
     std::vector<wrapper::Shader> m_shaders;
     std::vector<wrapper::Texture> m_textures;
     std::vector<wrapper::UniformBuffer> m_uniform_buffers;
     std::vector<wrapper::ResourceDescriptor> m_descriptors;
-
-    // NOTE: We use unique_ptr for easy frame graph recreation during swapchain invalidation
-    std::unique_ptr<FrameGraph> m_frame_graph;
-
     std::vector<OctreeGpuVertex> m_octree_vertices;
     std::vector<std::uint16_t> m_octree_indices;
 
     void setup_frame_graph();
     void generate_octree_indices();
 
+    void create_imgui_overlay();
```

Comment:
We should maybe name it `create_gui_overlay()` i would not, use imgui everywhere but just generic gui. ImGUI is just our tool.

Reply:
I inlined it and removed `create_imgui_overlay` entirely, see yeetari's review.
Done.